featurebyte/worker/util/batch_feature_creator.py

```diff
@@ -0,0 +1,264 @@
+"""
+Batch feature creator
+"""
+from typing import Dict, List, Sequence, Set, Union
+
+from bson import ObjectId
+
+from featurebyte.common.progress import get_ranged_progress_callback
+from featurebyte.enum import ConflictResolution
+from featurebyte.exception import DocumentInconsistencyError
+from featurebyte.logging import get_logger
+from featurebyte.models import FeatureModel
+from featurebyte.models.base import PydanticObjectId
+from featurebyte.query_graph.graph import QueryGraph
+from featurebyte.routes.feature.controller import FeatureController
+from featurebyte.schema.feature import BatchFeatureItem, FeatureServiceCreate
+from featurebyte.schema.worker.task.batch_feature_create import BatchFeatureCreateTaskPayload
+from featurebyte.schema.worker.task.feature_list_batch_feature_create import (
+    FeatureListCreateWithBatchFeatureCreationTaskPayload,
+)
+from featurebyte.service.feature import FeatureService
+from featurebyte.service.feature_namespace import FeatureNamespaceService
+from featurebyte.service.namespace_handler import NamespaceHandler
+from featurebyte.worker.task.batch_feature_create import execute_sdk_code
+from featurebyte.worker.util.task_progress_updater import TaskProgressUpdater
+
+logger = get_logger(__name__)
+
+
+class BatchFeatureCreator:
```

Comment:
we extract this out so that it can be injected into the 2 call sites. this will allow us to flatten the inheritance hierarchy as well and solve some typing issues